# **Predicting Employee Absenteeism From Work**

# Abstract
The database was created with records of absenteeism at work from July 2007 to July 2010 at a courier company in Brazil

Absenteeism is a habitual pattern of absence from a duty or obligation without good reason. Generally, absenteeism is unplanned absences. If a workplace exhibits a high degree of absenteeism there is a problem. It has been viewed as an indicator of poor individual performance, as well as a breach of an implicit contract between employee and employer.

# **Problem Statement**
Today's work environment has a lot of problems including:


*   Managers setting unachievable goals
*   Competition among industries
*   high risk of being unemployed

As a result, employees are likely to face huge pressure and work stress which has effect on employee health




**Aim of the portfolio**

To predict hours of Absenteeism at work using the features provided in the data set with information like:
* How far they live from their workplace.
* How many children and pets they have.
* Do they have higher education?

**Data source**
UCI Machine Learning Repository

**Basic Imports**

In [74]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")


In [75]:
import os

cwd = os.getcwd()  # Get the current working directory (cwd)
files = os.listdir(cwd)  # Get all the files in that directory
print("Files in %r: %s" % (cwd, files))

Files in 'e:\\userportal\\vinnie\\My-Portfolio': ['.git', '.vscode', 'Absenteeism_at_work.csv', 'Absenteeism_at_work.xls', 'predicting_employee_absenteeism.ipynb', 'README.md']


In [76]:
pip install xlrd


# Loading the data set

In [77]:
Data = pd.read_excel('E:\\userportal\\vinnie\\My-Portfolio\\Absenteeism_at_work.xls')
Data.head() #visualizing the first and last few rows


ID  Reason for absence  ...  Body mass index  Absenteeism time in hours
0  11                  26  ...               30                          4
1  36                   0  ...               31                          0
2   3                  23  ...               31                          2
3   7                   7  ...               24                          4
4  11                  23  ...               30                          2

[5 rows x 21 columns]

# Basic Analysis

In [78]:
Data.tail()

ID  Reason for absence  ...  Body mass index  Absenteeism time in hours
735  11                  14  ...               30                          8
736   1                  11  ...               29                          4
737   4                   0  ...               34                          0
738   8                   0  ...               35                          0
739  35                   0  ...               25                          0

[5 rows x 21 columns]

In [79]:
round(Data.describe())

ID  Reason for absence  ...  Body mass index  Absenteeism time in hours
count  740.0               740.0  ...            740.0                      740.0
mean    18.0                19.0  ...             27.0                        7.0
std     11.0                 8.0  ...              4.0                       13.0
min      1.0                 0.0  ...             19.0                        0.0
25%      9.0                13.0  ...             24.0                        2.0
50%     18.0                23.0  ...             25.0                        3.0
75%     28.0                26.0  ...             31.0                        8.0
max     36.0                28.0  ...             38.0                      120.0

[8 rows x 21 columns]

In [80]:
Data.describe(include=np.number)

ID  ...  Absenteeism time in hours
count  740.000000  ...                 740.000000
mean    18.017568  ...                   6.924324
std     11.021247  ...                  13.330998
min      1.000000  ...                   0.000000
25%      9.000000  ...                   2.000000
50%     18.000000  ...                   3.000000
75%     28.000000  ...                   8.000000
max     36.000000  ...                 120.000000

[8 rows x 21 columns]

In [81]:
#get the names of the columns
Data.columns

Index(['ID', 'Reason for absence', 'Month of absence', 'Day of the week',
       'Seasons', 'Transportation expense', 'Distance from Residence to Work',
       'Service time', 'Age', 'Work load Average/day ', 'Hit target',
       'Disciplinary failure', 'Education', 'Son', 'Social drinker',
       'Social smoker', 'Pet', 'Weight', 'Height', 'Body mass index',
       'Absenteeism time in hours'],
      dtype='object')

In [82]:
Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 740 entries, 0 to 739
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype
---  ------                           --------------  -----
 0   ID                               740 non-null    int64
 1   Reason for absence               740 non-null    int64
 2   Month of absence                 740 non-null    int64
 3   Day of the week                  740 non-null    int64
 4   Seasons                          740 non-null    int64
 5   Transportation expense           740 non-null    int64
 6   Distance from Residence to Work  740 non-null    int64
 7   Service time                     740 non-null    int64
 8   Age                              740 non-null    int64
 9   Work load Average/day            740 non-null    int64
 10  Hit target                       740 non-null    int64
 11  Disciplinary failure             740 non-null    int64
 12  Education                        740 non-null    i

Pandas profiling - make EDA easier with single line of code

In [83]:
pip install https://github.com/pandas-profiling/pandas-profiling/archive/master.zip

  Using cached https://github.com/pandas-profiling/pandas-profiling/archive/master.zip (25.9 MB)

  Created wheel for pandas-profiling: filename=pandas_profiling-3.1.1-py2.py3-none-any.whl size=261277 sha256=bba8ea1cc0792f761c5e26faa84bfedd958974570cb4d2ebc74a2ff1ca79170a
  Stored in directory: C:\Users\admin\AppData\Local\Temp\pip-ephem-wheel-cache-yp239se7\wheels\64\b6\85\dfc808b23666a5910371784e349d28818006ff63ed9cfeca59
Successfully built pandas-profiling


In [84]:
from pandas_profiling import ProfileReport

In [85]:
profile = ProfileReport(Data, title = "Panas Profile Report")

profile.to_widgets() #generate report 

Generate report structure: 100%|██████████| 1/1 [00:13<00:00, 13.70s/it]


In [86]:
#Summary data cleaning - figurin out in a table
checking = []

for i in Data.columns:
    checking.append([
        i,
        Data[i].dtypes,
        Data[i].isna().sum(),
        round((((Data[i].isna().sum() )/ len(Data)) * 100),2),
        Data[i].nunique(),
        Data[i].drop_duplicates().sample(2).values
    ])
pd.DataFrame(data = checking,
            columns = [
                'Features',
                'Data_type',
                'Null',
                'Null_percentage',
                'Unique',
                'Unique_sample'
            ])

Features Data_type  ...  Unique     Unique_sample
0                                ID     int64  ...      36          [16, 35]
1                Reason for absence     int64  ...      28          [19, 24]
2                  Month of absence     int64  ...      13            [1, 8]
3                   Day of the week     int64  ...       5            [3, 6]
4                           Seasons     int64  ...       4            [3, 2]
5            Transportation expense     int64  ...      24        [289, 179]
6   Distance from Residence to Work     int64  ...      25          [45, 15]
7                      Service time     int64  ...      18          [14, 12]
8                               Age     int64  ...      22          [33, 41]
9            Work load Average/day      int64  ...      38  [284853, 264249]
10                       Hit target     int64  ...      13          [96, 91]
11             Disciplinary failure     int64  ...       2            [0, 1]
12                        Education     int64  ...       4            [1, 3]
13                              Son     int64  ...       5            [4, 1]
14                   Social drinker     int64  ...       2            [0, 1]
15                    Social smoker     int64  ...       2            [1, 0]
16                              Pet     int64  ...       6            [4, 2]
17                           Weight     int64  ...      26          [77, 94]
18                           Height     int64  ...      14        [170, 178]
19                  Body mass index     int64  ...      17          [30, 33]
20        Absenteeism time in hours     int64  ...      19           [0, 48]

[21 rows x 6 columns]

In [ ]:
Data.isna()

In [ ]:
sns.heatmap(Data.isnull())

<AxesSubplot:>

As seen in the above summary there are no null values and there will be no steps to impute or delete some rows or columns

# Exploratory Data analysis

In [ ]:
#Get the unique values values with their maximum and minimum 
for column in Data.columns:
    print(f"===============Column: {column}==============")
    print(f"Number of unique values: {Data[column].nunique()}")
    print(f"Max: {Data[column].max()}")
    print(f"Min: {Data[column].min()}")

===============Column: ID==============
Number of unique values: 36
Max: 36
Min: 1
===============Column: Reason for absence==============
Number of unique values: 28
Max: 28
Min: 0
===============Column: Month of absence==============
Number of unique values: 13
Max: 12
Min: 0
===============Column: Day of the week==============
Number of unique values: 5
Max: 6
Min: 2
===============Column: Seasons==============
Number of unique values: 4
Max: 4
Min: 1
===============Column: Transportation expense==============
Number of unique values: 24
Max: 388
Min: 118
===============Column: Distance from Residence to Work==============
Number of unique values: 25
Max: 52
Min: 5
===============Column: Service time==============
Number of unique values: 18
Max: 29
Min: 1
===============Column: Age==============
Number of unique values: 22
Max: 58
Min: 27
===============Column: Work load Average/day ==============
Number of unique values: 38
Max: 378884
Min: 205917
===============Column: Hit target

In [ ]:
#shape of the data set
print("The shape of the data is:{}" .format(Data.shape))

The shape of the data is:(740, 21)


In [ ]:
#Data types
print('Data types of the data set is:{}'.format(Data.dtypes))

Data types of the data set is:ID                                 int64
Reason for absence                 int64
Month of absence                   int64
Day of the week                    int64
Seasons                            int64
Transportation expense             int64
Distance from Residence to Work    int64
Service time                       int64
Age                                int64
Work load Average/day              int64
Hit target                         int64
Disciplinary failure               int64
Education                          int64
Son                                int64
Social drinker                     int64
Social smoker                      int64
Pet                                int64
Weight                             int64
Height                             int64
Body mass index                    int64
Absenteeism time in hours          int64
dtype: object


In [ ]:
#Duplicated values
Data.duplicated().sum()

34

In [ ]:
Data[Data.duplicated()]

ID  Reason for absence  ...  Body mass index  Absenteeism time in hours
16    3                  23  ...               31                          2
68   34                  23  ...               28                          3
88   28                  23  ...               24                          1
109  10                  22  ...               27                          8
127  34                  27  ...               28                          2
128  34                  27  ...               28                          2
131  34                  27  ...               28                          2
132  27                  23  ...               21                          2
133  34                  27  ...               28                          2
305   5                  23  ...               38                          2
351   3                  28  ...               31                          1
376   3                  27  ...               31                          3
385   3                  27  ...               31                          3
386   3                  27  ...               31                          3
388   3                  27  ...               31                          3
389   3                  27  ...               31                          3
440  22                  23  ...               19                          3
477  24                  28  ...               23                          2
496  24                  28  ...               23                          1
560  28                  23  ...               24                          3
605   3                  27  ...               31                          2
607   3                  27  ...               31                          2
610   3                  27  ...               31                          2
615   3                  27  ...               31                          2
616   3                  27  ...               31                          2
631   3                  27  ...               31                          2
632   3                  27  ...               31                          3
641   3                  27  ...               31                          2
643   3                  27  ...               31                          3
666  22                  27  ...               19                          2
669  22                  27  ...               19                          2
673  22                  27  ...               19                          2
699  15                  28  ...               25                          2
700  22                  27  ...               19                          2

[34 rows x 21 columns]

In [ ]:
Data = Data.drop_duplicates(keep="first").reset_index()

In [ ]:
Data.shape

(706, 22)

In [ ]:
#Check the number of employees in the company using their unique IDs
Data["ID"].unique()

array([11, 36,  3,  7, 10, 20, 14,  1, 24,  6, 33, 18, 30,  2, 19, 27, 34,
        5, 15, 29, 28, 13, 22, 17, 31, 23, 32,  9, 26, 21,  8, 25, 12, 16,
        4, 35], dtype=int64)

We have 35 employees

However this column does not carry any numeric information but its only a label variable (a number that is there to distinguish the individuals from one another, not to carry any numeric information) same as index

# Remove Irrelevant faetures

In [ ]:
Data.drop(Data["ID"], inplace=True)

# Remove features if they have more than 80% correlation

In [ ]:
def correlation(Data, threshold):
    col_corr = set()  # Set of all the names of correlated columns
    corr_matrix = Data.corr()
    for i in range(len(corr_matrix.columns)):
        for j in range(i):
            if abs(corr_matrix.iloc[i, j]) > threshold: # we are interested in absolute coeff value
                colname = corr_matrix.columns[i]  # getting the name of column
                col_corr.add(colname)
    return col_corr


corr_features = correlation(Data, 0.9)
len(set(corr_features))

1

In [ ]:
corr_features


{'Body mass index'}

We therefore drop Body mass index since it has correlation more than 90%

In [ ]:
Data.drop(['Body mass index'], axis=1, inplace=True)

Pet is more objective than subjective.
Month of absence, Day of the week', 'Seasons', we are looking for the number of absences and not a period of absences.
Weight,Height',are very subjective data, for our analysis it will not influence the number of absences
we are not analyzing reasons for absences hence we can also delete reasons for absence
Month of absence, Day of the week', 'Seasons', we are looking for the number of absences and not a period of absences;

In [ ]:
#Data.drop(['Month of absence','Seasons','Reason for absence','Month of absence','Hit target','Pet', 'Weight','Height'], axis=1, inplace=True )

In [ ]:
Data["Absenteeism time in hours"].unique()

array([  4,   8,   2,   3,  32,   0,   1,   5,  16,  40,  24,  64,  56,
        80, 120, 112, 104,  48], dtype=int64)

In [ ]:
#Average time of absenteeism at work
Data["Absenteeism time in hours"].mean()

7.152238805970149

In [ ]:
Data[Data['Day of the week']==7]['Absenteeism time in hours'].mean()

nan

Observe reasons for absence

In [ ]:
Data["Absenteeism time in hours"].unique()

array([  4,   8,   2,   3,  32,   0,   1,   5,  16,  40,  24,  64,  56,
        80, 120, 112, 104,  48], dtype=int64)

- 0: 'Unknown',
- 2: 'Neoplasms',
- 3: 'Diseases of the blood and blood-forming organs and certain disorders involving the immune mechanism',
- 4: 'Endocrine, nutritional and metabolic diseases',
- 5: 'Mental and behavioural disorders',
- 6: 'Diseases of the nervous system',
- 7: 'Diseases of the eye and adnexa',
- 8: 'Diseases of the ear and mastoid process',
- 9: 'Diseases of the circulatory system',
- 10: 'Diseases of the respiratory system',
- 11: 'Diseases of the digestive system',
- 12: 'Diseases of the skin and subcutaneous tissue',
- 13: 'Diseases of the musculoskeletal system and connective tissue',
- 14: 'Diseases of the genitourinary system',
- 15: 'Pregnancy, childbirth and the puerperium',
- 16: 'Certain conditions originating in the perinatal period',
- 17: 'Congenital malformations, deformations and chromosomal abnormalities',
- 18: 'Symptoms, signs and abnormal clinical and laboratory findings, not elsewhere classified',
- 19: 'Injury, poisoning and certain other consequences of external causes',
- 20: 'External causes of morbidity and mortality',
- 21: 'Factors influencing health status and contact with health services',
- 22: 'Patient follow-up',
- 23: 'Medical consultation',
- 24: 'Blood donation',
- 25: 'Laboratory examination',
- 26: 'Unjustified absence',
- 27: 'Physiotherapy',
- 28: 'Dental consultation'


In [ ]:
# plot distribution of "reason for absence" column
sns.set_style("whitegrid")
sns.catplot(data=Data, x='Reason for absence', kind= 'count',size=4, aspect=4, palette='muted')

In [ ]:
mean_reason = Data[['Reason for absence', 'Absenteeism time in hours']].groupby('Reason for absence').mean().reset_index().sort_values('Absenteeism time in hours', ascending = False)
plt.figure(figsize = (20,5))
sns.barplot(mean_reason['Reason for absence'],   mean_reason['Absenteeism time in hours'])
plt.title('mean absent hours based on reason of absence')

Text(0.5, 1.0, 'mean absent hours based on reason of absence')

The highest mean hours of absensce is because of disease of circulatory system. It is related to heart disease, based on the data heart disease took longer time to recover, it is 18 hours longer than the top 2 absent reason which is neoplasm.

In [ ]:
month = Data[['Month of absence', 'Absenteeism time in hours']].groupby('Month of absence').sum().reset_index()
plt.figure(figsize = (10,5))
sns.lineplot(x = month['Month of absence'], y= month['Absenteeism time in hours'])

<AxesSubplot:xlabel='Month of absence', ylabel='Absenteeism time in hours'>

In march and july employee tends to go absent.

In [ ]:
day = Data[['Day of the week', 'Absenteeism time in hours']].groupby('Day of the week').sum().reset_index()
day
plt.figure(figsize = (10,5))
sns.lineplot(x = day['Day of the week'], y= day['Absenteeism time in hours'])

<AxesSubplot:xlabel='Day of the week', ylabel='Absenteeism time in hours'>

There were no data that recorded on monday (0 records).
The more closer to weekend, the lower the total absenteeism hours

In [ ]:
sns.scatterplot(Data['Transportation expense'], Data['Absenteeism time in hours'])

<AxesSubplot:xlabel='Day of the week', ylabel='Absenteeism time in hours'>

Checking the scatter plot, the were no linear relationship between transporation expense and absenteeism time in hours, additional, the confusion matrix there is weak correlation.

In [ ]:
sns.scatterplot(Data['Distance from Residence to Work'], Data['Absenteeism time in hours'])

<AxesSubplot:xlabel='Day of the week', ylabel='Absenteeism time in hours'>

In [ ]:
sns.heatmap(Data[['Distance from Residence to Work', 'Absenteeism time in hours']].corr('spearman'), annot = True)

<AxesSubplot:>

Checking the scatter plot, the were no linear relationship between transporation expense and absenteeism time in hours.Based on confusion matrix, there are positive correlation but the number is indicating that it is very weak because the number is very close to 0. hence we can conclude that Distance did not contribute to absenteeism in work.

In [ ]:
sns.scatterplot(Data['Service time'], Data['Absenteeism time in hours'])

<AxesSubplot:xlabel='Service time', ylabel='Absenteeism time in hours'>

In [ ]:
sns.heatmap(Data[['Service time', 'Absenteeism time in hours']].corr('spearman'), annot = True)

<AxesSubplot:>

 The were no linear relationship between service time and absenteeism time in hours. Based on confusion matrix, there are negative correlation but the number is indicating that it is very weak because the number is very close to 0. We could conclude that service time is not the factors that made the employee absent.

In [ ]:
sns.scatterplot(Data['Age'], Data['Absenteeism time in hours'])

<AxesSubplot:xlabel='Age', ylabel='Absenteeism time in hours'>

In [ ]:
sns.heatmap(Data[['Age', 'Absenteeism time in hours']].corr('spearman'), annot = True)

<AxesSubplot:>

The were no linear relationship between age and absenteeism time in hours. Based on confusion matrix, there are negative correlation but the number is indicating that it is very weak because the number is very close to 0. We could conclude that age is not the factors that made the employee absent. Employee that is on its senior age does not mean that they are more prone to sickness, also employee that are still young does not mean they're healthy.

In [ ]:
Data[['Disciplinary failure', 'Absenteeism time in hours']].groupby('Disciplinary failure').sum()

Absenteeism time in hours
Disciplinary failure                           
0                                          4792
1                                             0

The employee that are absent does not because of indisciplinary action
the indisiciplinary action is not because of they are absent with no reason. but it is more indiciplinary action in work environment

In [ ]:
Data[['Education', 'Absenteeism time in hours']].groupby('Education').count()

Absenteeism time in hours
Education                           
1                                549
2                                 45
3                                 72
4                                  4

In [ ]:
education= Data[['Education', 'Absenteeism time in hours']].groupby('Education').sum().reset_index()
sns.barplot(education['Education'], education['Absenteeism time in hours'])

<AxesSubplot:xlabel='Education', ylabel='Absenteeism time in hours'>

The majority education of employee is on highschool level and the majority of absenteeism are from high-school education - employee.

In [ ]:
son= Data[['Son', 'Absenteeism time in hours']].groupby('Son').sum().reset_index()
sns.barplot(son['Son'], son['Absenteeism time in hours'])

<AxesSubplot:xlabel='Son', ylabel='Absenteeism time in hours'>

The more the number of sons, the lesser  the numbers of being absent in work

In [ ]:
Data[['Son', 'Absenteeism time in hours']].corr('spearman')

Son  Absenteeism time in hours
Son                        1.000000                   0.122385
Absenteeism time in hours  0.122385                   1.000000

However, there is no relationship betweeen number of sons and absenteeism time in hours

# Absenteeism time wise

In [ ]:

plt.figure(figsize = (15,6))
sns.barplot(Data.groupby(['Reason for absence'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Reason for absence'])['Absenteeism time in hours'].sum(), palette="icefire")

<AxesSubplot:xlabel='Reason for absence', ylabel='Absenteeism time in hours'>

# Absenteeism  Time Per Day of The Week And Per Month 

In [ ]:
plt.figure(figsize = (15,4))
plt.subplot(1, 2, 1)
sns.barplot(Data.groupby(['Day of the week'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Day of the week'])['Absenteeism time in hours'].sum(), palette="magma")
plt.subplot(1, 2, 2)
sns.barplot(Data.groupby(['Month of absence'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Month of absence'])['Absenteeism time in hours'].sum(), palette="magma")

<AxesSubplot:xlabel='Month of absence', ylabel='Absenteeism time in hours'>

# Absenteeism time - Age and BMI

In [ ]:
plt.figure(figsize = (16,4))
plt.subplot(1, 2, 1)
sns.barplot(Data.groupby(['Height'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Height'])['Absenteeism time in hours'].sum(), palette="Paired")
plt.subplot(1, 2, 2)
sns.barplot(Data.groupby(['Weight'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Weight'])['Absenteeism time in hours'].sum(), palette="Paired")

<AxesSubplot:xlabel='Weight', ylabel='Absenteeism time in hours'>

# Absenteeism Education Wise

In [ ]:
plt.figure(figsize = (10,5))

sns.barplot(Data.groupby(['Education'])['Absenteeism time in hours'].sum().index, 
            Data.groupby(['Education'])['Absenteeism time in hours'].sum(), palette="Set2")

<AxesSubplot:xlabel='Education', ylabel='Absenteeism time in hours'>

In [ ]:
#get the relationship between absenteeism time in hours and seasons
sns.jointplot(x=Data["Absenteeism time in hours"], y=Data["Seasons"],data=Data)

In [ ]:
#get the relationship between absenteeism time in hours and Age
sns.jointplot(x=Data["Absenteeism time in hours"], y=Data["Age"],data=Data)

In [ ]:
plt.figure(figsize=(12,6))
sns.lmplot(x='Age',y='Absenteeism time in hours',data=Data,hue='Day of the week',size=5,aspect=2)

Separate numerical and categorical data

In [ ]:
categorical_vars = ['Reason for absence','Month of absence','Day of the week',
                     'Seasons','Disciplinary failure', 'Education', 'Social drinker',
                     'Social smoker', 'Pet', 'Son']
numerical_vars = Data.drop(columns=categorical_vars).columns.to_list()

In [ ]:
# number of unique values in categorical atts
Data[categorical_vars].nunique()

Reason for absence      28
Month of absence        13
Day of the week          5
Seasons                  4
Disciplinary failure     2
Education                4
Social drinker           2
Social smoker            2
Pet                      6
Son                      5
dtype: int64

In [ ]:
# plot distribution of numerical attributes
num_data = Data[numerical_vars]
plt.figure(figsize=(20,20))
for i in range(1, 11):
    plt.subplot(3, 4, i)
    sns.distplot(num_data[num_data.columns[i-1]],bins=14)

In [ ]:
# correlation matrix
num_corr = num_data.corr()
plt.figure(figsize=(10,6))
sns.heatmap(num_corr, annot=True, fmt=".3f",vmin=-1, vmax=1, linewidths=.5, cmap = sns.color_palette("RdBu", 100))
plt.yticks(rotation=0)
plt.show()

In [ ]:
len(Data)

670

Outlier detection 

In [ ]:
# Check for outliers using boxplots and drop them
for num_at in numerical_vars:
    # Getting 75 and 25 percentile of variable "i"
    Q3, Q1 = np.percentile(Data[num_at], [75,25])
    MEAN = Data[num_at].mean()
    
    # Calculating Interquartile range
    IQR = Q3 - Q1
    
    # Calculating upper extremes and lower extremes
    minimum = Q1 - (IQR*1.5)
    maximum = Q3 + (IQR*1.5)
    
    # Replacing all the outliers value to Mean
    data_clean=Data.drop(Data.loc[Data[num_at]< minimum,num_at].index) 
    data_clean=Data.drop(Data.loc[Data[num_at]> maximum,num_at].index)

In [ ]:
len(data_clean)

628

Length reduce from 670 to 628

Normalization

Minority = not absent majority = absent 1 = absent (not present) 0 = not absent (present) True positive = Actual absent, prediction absent True negative = Actual not absent, prediction not absent False positive = Actual not absent (present), but the prediction is absent False negative = Actual absent but the prediction is present -> main concern

MAIN CONCERN

our main concern is to supress the number of false negative while also having least number of false positive. (balancing)
Why we want to balance out the false positive and false negative. Case: After detecting the employee that tends to go sick. the company apporach is to examine the employees health to mental-wellness-program and healthy-program. Also taking several health examination. If we have big false positive then company have to pay more for the program and examination (more people that detected tends to go absent but the actual he is healthy). Also we need to supress the false negative so that our prediction is more likely to be accurate in predicting people that tends to go absent.

In [ ]:
ab = []
for i in Data['Absenteeism time in hours']:
    if i > 0 :
        ab.append(1)
    else:
        ab.append(0)
Data['Absenteeism'] = ab #classification target

In [ ]:
Data.drop(columns = ['Reason for absence', 'Absenteeism time in hours','Month of absence', 'Day of the week'], inplace = True)

KeyError: "['Reason for absence' 'Absenteeism time in hours' 'Month of absence'\n 'Day of the week'] not found in axis"

Modelling

In [ ]:
X = Data.drop(columns = 'Absenteeism')
y = Data['Absenteeism']
X_train, X_test, y_train, y_test = train_test_split(X,y, stratify = y , test_size = 0.3, random_state =42)

NameError: name 'train_test_split' is not defined